### PR TITLE
[Unity][Transform] Allow static Relax arguments to dynamic PrimFunc

### PR DESCRIPTION
**Pull request by Lunderberg**
Prior to this commit, the `relax.transform.FuseTIR` transform required that the shapes arguments passed into a `PrimFunc` be structurally equivalent to the shapes of the parameters, and that any replacement of symbolic `tir.Var` be with a symbolic `tir.Var` in the fused function.

This commit updates the `SymbolicMatcher` to instead extract a `Map<tir::Var, PrimExpr>`.  As a result, a Relax tensor with statically-known shape can be passed into a TIR PrimFunc with dynamic shape.  The resulting fused TIR function is in terms of the statically-known shape, and no longer contains the symbolic variable.

### CODE REVIEW

**Comment by masahi** [comment]
cc @Hzfengsy

File: src/relax/transform/fuse_tir.cc [hunk 1]
```diff
@@ -39,31 +39,41 @@ namespace tir {
...
-    return matched && ExprFunctor::VisitExpr(n, other);
+  bool VisitExpr(const PrimExpr& node, const PrimExpr& other) {
+    if (node.same_as(other)) {
+      return true;
+    } else if (node.dtype().code() != other.dtype().code()) {
+      return false;
+    } else {
+      return ExprFunctor::VisitExpr(node, other);
+    }
+    // bool matched = node.same_as(other) || ((node->type_index() == other->type_index()) &&
```
Comment (Hzfengsy) [inline src/relax/transform/fuse_tir.cc:65]: Please remove useless lines

**Review by Hzfengsy — APPROVED** [review]
General LGTM.

File: src/relax/transform/fuse_tir.cc (hunk 1 above)
Comment (Lunderberg) [reply src/relax/transform/fuse_tir.cc]: Thank you for the catch, and removed.

**Pull request merged**